tools/flash_shelly.sh

```diff
@@ -0,0 +1,201 @@
+#!/bin/bash
+#
+#  Copyright (c) 2020 Andrew Blackburn
+#  All rights reserved
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+#  This script will probe for any shelly device on the network and it will
+#  attemp to update them to the lastest firmware version availible.
+#  This script will not flash any firmware to a device that is not already on a
+#  version of this firmware, if you are looking to flash your device from stock
+#  or any other firmware please follow instructions here:
+#  https://github.com/mongoose-os-apps/shelly-homekit/blob/master/README.md
+#
+#  -u, --update        Update device(s) to the lastest available firmware.
+#  -c, --check-only    Only check for updates.
+#  -h, --help          This help text.
+#
+#  usage: ./flash_shelly.sh -u
```

Comment:
i think we can get rid of `-c` and make it the default behavior unless `-u` is given.
and i'd like to see `-a` to request that update be done to all devices.

if neither `-a` nor specific device is given, output help.

Reply:
ok so make -u default and -a for auto.

Reply:
hang on don't quite understand, if you remove -c and make it default ie no flag or device that will run a check for all devices on network, why would you want help shown?

Reply:
or do you mean if 
 `./flash_shelly.sh xxxxxxx.local`. it would perform a check on that single device
`./flash_shelly.sh -a` would perform a check on `all` devices
`./flash_shelly.sh -u xxxxxx.local` would update that device
`./flash_shelly.sh -u -a` would update all devices 
`./flash_shelly.sh` would show help ?
is that what you meant ?

Reply:
ok, thinking aloud here. the script is called flash_shelly, so it makes sense that if you invoke it you want to flash by default, so `-u` seems superfluous. how about, instead, we do this:
0) when invoked without arguments or with `-h`, display help
1) when invoked with with one or more arguments flashes latest homekit firmware to those devices. thus, `./flash_shelly.sh xxx yyy zzz` flashes latest firmware to xxx, yyy and zzz. if running stock firmware, convert, if already running homekit fw - update if necessary.
2) if invoked with `-a`, do it to all the devices on the network
3) if invoked with `-n`, go through the motions but do not actually flash - print "would have converted to homekit foirmware" or "would have updated from a.b.c to x.y.z" but don't actually do it. this can be used with explicit device list as in (1) or with `-a` (i.e. `-a -n`).
4) if invoked with `-l`, simply list devices and their current versions (stock / homekit, current version)

perhaps, in the future we can also add `-r` - revert to stock firmware, either with explicit device list or `-a` (no need to do it in this iteration, we can do it in a future change).